mentoring/pairing/tests.py

```diff
@@ -53,6 +55,23 @@ def test_model_make_pair(self):
 
         self.assertTrue(HistoricalPair.already_paired(p))
 
+    def test_make_pair_rest(self):
```

Comment:
I was looking for a ``mentor != learner`` test, but it looks like the UI enforces that at the moment. Model-level enforcement of ``learner != mentor`` is tricky. The method [clean](https://docs.djangoproject.com/en/3.1/ref/models/instances/#django.db.models.Model.clean) can have the logic, and will be called when you save objects in forms like the Django admin. Django Rest Framework does not call  ``clean()`` or ``.full_clean()``, and wants [object-level validation](https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation) in the ``.validate()`` method of the serializer. ``clean()`` and ``full_clean()`` are not called when calling ``save`` directly in code, or when doing bulk updates. If you want model-level safety everywhere, you end up with a lot of duplicate code.

Reply:
Hm, this is something I hadn't though to try to prevent.  It's probably not critical (nothing will explode if a participant is paired with themselves).  Filed #47.